### ISSUE TITLE
Stratified cell training

### ISSUE THREAD

**Issue by BradBalderson** [gagneurlab/scooby#18]
Hi

Working on training scooby on a new dataset of cells. I want to circumvent the longer training time, and preferably only use 1 A40 GPU. 

Aside from utilizing Flashzoi as you show in thread #12, I am thinking one good way would be to randomly sample an equal number of cells from each of my annotated cell types, so the total training set of cells will be ~12,000 cells / ~500,000 (inspired by geosketch, Hie et al).

Wondering if you guys have considered stratified training like this? 

And also if you think this will actually result in a speed boost or not? (i.e. is the number of cells a major bottle-neck or is it sequence loading speed, for example.)

In terms of GPU memory usage, did you need the 8 A40 GPUs in paper because of the size of Borzoi? i.e. just wondering if having a smaller set of cells would allow for single A40 GPU training.

Thankyou for this awesome model

**Comment by johahi** [comment]
Hi Brad, 

We experimented with cell sampling early in the project, but back then did not find it to greatly improve predictive performance (it could be that this finding was still on the incorrect train-test-split though). 
The number of cells is not really a scaling issue, as scooby does not really scale with that - per sequence, you subsample a constant amount of cells during training, for now we did not have to change that number nor train for longer, even when training a model on 1.2M cells - scooby will "work". Although we did find it beneficial to train longer in general, in terms of sequences seen.
Regarding the 8 A40s, we mainly did this to speed up training, but you might as well train on 1 A40 with batch size up to 2 or 4 I'd assume, but you might have to adapt the learning rate for this.
